src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/BatchingSqlJournal.cs

```diff
@@ -1070,22 +1070,29 @@ protected virtual async Task HandleReplayAllMessages(ReplayAllEvents req, TComma
 
             try
             {
-                var maxSequenceNr = 0L;
+                var toOffset = req.ToOffset;
                 var fromOffset = req.FromOffset;
+                var max = req.Max;
+
+                var take = Math.Min(toOffset - fromOffset, max);
+
+                command.CommandText = HighestOrderingSql;
+                command.Parameters.Clear();
+
+                var maxOrdering = (await command.ExecuteScalarAsync()) as long? ?? 0L;
```

Comment:
So we have to query the max ordering number every time? Can't we just add an offset to the previous value on the command?

Reply:
These are legacy steps taken by old codes that does paging against a database. 
Basically, the publisher would have a variable called ToOffset which is initialized as long.MaxValue. The first executor result will contain the highest sequence number at that query. 
The publisher compares the value with ToOffset, if it is less than ToOffset, then store it in ToOffset.
Use the value of ToOffset in subsequent requests as a limit.

I don't know why it was done this way, I was just avoiding possible run-time behavior changes

Reply:
So do we only query that value once or do we have to do it for each piece of pagination?

Reply:
Basically, that MaxOrderingNr return value is only used in the `Current` query variant to limit returned events to only the last event row when the query was first executed

Reply:
Ah got it - that makes perfect sense